lib/push.py

```diff
@@ -37,16 +38,18 @@ def main(cli):
 
 class Image_Pusher:
 
-   # Note; We use functions to create the blank config and manifest to to
+   # Note; We use functions to create the blank config and manifest to
    # avoid copy/deepcopy complexity from just copying a default dict.
 
-   __slots__ = ("config",    # sequence of bytes
-                "dst_ref",   # destination of upload
-                "image",     # Image object we are uploading
-                "layers",    # list of (digest, .tar.gz path), lowest first
-                "manifest")  # sequence of bytes
+   __slots__ = ("cache_upload",  # bool if upload cache is enabled
```

Comment:
Adding an attribute, in this case a boolean, to augment behavior is certainly one approach. Consider class inheritance and how that may change your logic. For example:

```python
class Image_Pusher:
[...]
class Image_Pusher_Cache_Enabled (Image_Pusher):
```
`Image_Pusher_Cache_Enabled` can then augment methods of interest, e.g., `prepare(), `prepare_upload()`, `prepare_existing()`, etc. without relying on various `if self.cache_ulpoad` checks.